packages/openneuro-app/src/scripts/errors/errorBoundary.jsx

```diff
@@ -28,79 +101,61 @@ class ErrorBoundary extends React.Component {
       supportModal: false,
       error: props.error,
       eventId: null,
+      message: props.error ? props.error.message : '',
     }
   }
-
   static getDerivedStateFromError(error) {
-    return getDerivedStateFromErrorOnCondition(
-      error,
-      // error boundary should always be triggered in general case
-      () => true,
-    )
+    return getDerivedStateFromErrorOnCondition(error, () => true)
   }
-
   componentDidCatch(error) {
+    let message = this.state.message
+    if (!this.props.dataset) {
+      message = 'This dataset does not exist.'
+    } else if (this.props.dataset && this.props.dataset.snapshots.length < 1) {
+      message = 'This dataset has no associated snapshots.'
+    }
     Sentry.withScope(scope => {
       scope.setTag('datasetId', this.props.datasetId)
-      this.setState({ eventId: Sentry.captureException(error) })
+      this.setState({
+        eventId: Sentry.captureException(error),
+        message: message,
+      })
     })
   }
-
   closeSupportModal = () =>
     this.setState(prevState => ({
       ...prevState,
       supportModal: false,
     }))
-
   openSupportModalFromLink = e => {
     e.preventDefault()
     this.setState(prevState => ({
       ...prevState,
       supportModal: true,
     }))
   }
-
   render() {
-    const { supportModal } = this.state
+    const error = this.state.error || this.props.error
+    const { supportModal, message } = this.state
     const { subject, description } = this.props
-    const error = this.props.error || this.state.error
-    return this.state.hasError ? (
-      <>
-        <p className="generic-error-message">
-          {this.props.errorMessage || 'An error has occurred.'}
-          <br />
-          Please support us by documenting the issue with{' '}
-          <a onClick={this.openSupportModalFromLink}>
-            <u>FreshDesk</u>
-          </a>
-          .
-        </p>
-        {supportModal && (
-          <Overlay>
-            <ModalContainer>
-              <ExitButton onClick={this.closeSupportModal}>&times;</ExitButton>
-              <h3>Support</h3>
-              <hr />
-              <div>
-                To ensure that we can quickly help resolve this issue, please
-                provide as much detail as you can, including what you were
-                trying to accomplish when the error occurred.
-              </div>
-              <FreshdeskWidget
-                {...{
-                  subject,
-                  description,
-                  error,
-                  sentryId: this.state.eventId,
-                }}
-              />
-            </ModalContainer>
-          </Overlay>
-        )}
-      </>
-    ) : (
-      this.props.children
-    )
+    if (error) {
+      if (message) {
+        return <DatasetRedirect message={message} />
+      } else {
+        return (
+          <FreshdeskModal
+            error={message}
```

Comment:
message here is always falsy. error should just be the original error object (to be used by FreshdeskWidget).